### PR TITLE
move contract test out of component

### LINKED ISSUE #6454: remove otlphttpreceiver dependency from otlpexporter
Currently the `otlpexporter` has a dependency on `otlphttpreceiver` for testing purposes. This causes any package that includes the `otlpexporter` to also include the `otlphttpreceiver` unnecessarily.

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
remove dependency of otlphttpreceiver on otlpexporter

**Link to tracking Issue:**
Fixes #6454

### CODE REVIEW

File: internal/test/consume_contract_test.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
-package otlpexporter
+package test
```
Comment (bogdandrutu) [inline internal/test/consume_contract_test.go:4]: If we move this we create again the same circular dependency and even more:

otlpexporter -> main otel package (with internal)
otlpreceiver -> main otel package (with internal)

Now after this change `main otel package (with internal)` depends on `otlpexporter` and `otlpreceiver`

Comment (atoulme) [reply internal/test/consume_contract_test.go:4]: Can we create a new module for those tests? I was asking @codeboten if there is a package we can use

Comment (bogdandrutu) [reply internal/test/consume_contract_test.go:4]: We can definitely have some "e2e" module that we don't "export" (in internal).

Comment (atoulme) [reply internal/test/consume_contract_test.go:4]: Thanks! I'll get on that right away.

File: internal/e2e/go.mod [hunk 2]
```diff
@@ -0,0 +1,125 @@
+module go.opentelemetry.io/collector/internal/e2e
```
Comment (bogdandrutu) [inline internal/e2e/go.mod:1]: We should have it in the `versions.yaml` under ignored.

**Review by bogdandrutu — APPROVED** [review]

File: internal/test/consume_contract_test.go (hunk 1 above)
Comment (atoulme) [reply internal/test/consume_contract_test.go]: ok this is done, the diff is much better

**Review by codeboten — APPROVED** [review]
Nice! this is great, thanks @atoulme

**Comment by codeboten** [comment]
> Replace statements are out of date, please run "make crosslink" and commit the changes in this PR.

**Comment by atoulme** [comment]
good to go

**Pull request merged**